### ISSUE TITLE
Format code on demand or upon saving

### ISSUE THREAD

**Issue by kerscher** [lukexi/stack-ide-sublime#38]
Whenever the user calls a `Prettify this file` command, it should be possible to run the file through [stylish-haskell](http://haddock.stackage.org/lts-3.13/stylish-haskell-0.5.14.3/index.html), updating the buffer immediately.

Optionally, the user could leave an option set on preferences that whenever the file is saved it first passes through prettification.

Currently I leave a watch program doing this whenever the file changes.

The only downside is that `stylish-haskell` has to be installed, though it is on LTS.

**Comment by kerscher** [comment]
I was planning on adding this, but since there is the additional dependency I might create a separate package just for this command and link to here for a proper IDE.

On a side-note, now that `stack-ide` is [being deprecated](https://www.fpcomplete.com/blog/2015/10/new-haskell-ide-repo), what are the plans for the package?

**Comment by rvion** [comment]
@kerscher are you aware of https://github.com/haskell/haskell-ide-engine ?
This is probably the "future main backend" of all ide integrations :)

**Comment by lukexi** [comment]
Yep, I haven't kept up with how far along haskell-ide-engine is but I'm definitely planning to switch over to it as soon as it's ready to go!

**Comment by kerscher** [comment]
Thanks for the link @rvion, and for the heads-up @lukexi. I'll check the discussion there :)
